### ISSUE TITLE
trimap vs ivis? 

### ISSUE THREAD

**Issue by Hellisotherpeople** [eamid/trimap#8]
Can you do some comparisons between trimap and bearing researches ivis? I think that both of you use similar techniques (triplet networks). 

I'm curious to see what differes between each implementation.

**Comment by eamid** [comment]
I believe ivis trains a multilayer feed-forward network using a loss function on triplets. Therefore, ivis learns an intermediate representation using the dense layers before projecting the data into 2D.  Also, the loss function is defined as a hinge loss on the difference between the distances. Alternatively, TriMap directly minimizes the loss of the triplets, without learning an extra representation. Additionally, the loss is defined as a ratio in TriMap.

It is possible to augment TriMap to a Siamese network and learn a deep representation. This yields excellent results. However, it also requires extra computation and therefore, results in a longer training time. I am currently working on a JAX and I will add this feature in the future. I am also planning to add TriMap to the TF projector: https://projector.tensorflow.org/

**Comment by eamid** [comment]
I hope the comment above answers your question. Please feel free to contact me at: eamid@ucsc.edu if you have any further questions. I am going to close the issue for now.
